src/service/xhr-impl.js

```diff
@@ -162,262 +93,6 @@ export class Xhr {
-   *
-   * @param {JsonObject|string|undefined} response The structurally-cloneable
-   *     response to convert back to a regular Response.
-   * @param {string|undefined} responseType The original response type used to
-   *     initiate the XHR.
-   * @return {!FetchResponse|!Response} The deserialized regular response.
-   * @private
-   */
-  fromStructuredCloneable_(response, responseType) {
-    user().assert(isObject(response), 'Object expected: %s', response);
-
-    const isDocumentType = responseType == 'document';
-    if (typeof this.win.Response === 'function' && !isDocumentType) {
-      // Use native `Response` type if available for performance. If response
-      // type is `document`, we must fall back to `FetchResponse` polyfill
-      // because callers would then rely on the `responseXML` property being
-      // present, which is not supported by the Response type.
-      return new this.win.Response(response['body'], response['init']);
-    }
-
-    const lowercasedHeaders = map();
-    const data = {
-      status: 200,
-      statusText: 'OK',
-      responseText: (response['body'] ? String(response['body']) : ''),
-      /**
-       * @param {string} name
-       * @return {string}
-       */
-      getResponseHeader(name) {
-        return lowercasedHeaders[String(name).toLowerCase()] || null;
-      },
-    };
-
-    if (response['init']) {
-      const init = response['init'];
-      if (isArray(init.headers)) {
-        init.headers.forEach(entry => {
-          const headerName = entry[0];
-          const headerValue = entry[1];
-          lowercasedHeaders[String(headerName).toLowerCase()] =
-              String(headerValue);
-        });
-      }
-      if (init.status) {
-        data.status = parseInt(init.status, 10);
-      }
-      if (init.statusText) {
-        data.statusText = String(init.statusText);
-      }
-    }
-
-    if (isDocumentType) {
-      data.responseXML =
-          new DOMParser().parseFromString(data.responseText, 'text/html');
-    }
-
-    return new FetchResponse(data);
-  }
-
-  /**
-   * Performs the final initialization and requests the fetch. It does two
-   * main things:
-   * - It adds "__amp_source_origin" URL parameter with source origin
-   * - It verifies "AMP-Access-Control-Allow-Source-Origin" in the response
-   * USE WITH CAUTION: setting ampCors to false disables AMP source origin check
-   * but allows for caching resources cross pages.
-   *
-   * Note: requireAmpResponseSourceOrigin is deprecated. It defaults to
-   *   true. Use "ampCors: false" to disable AMP source origin check.
-   *
-   * @param {string} input
-   * @param {!FetchInitDef=} init
-   * @return {!Promise<!FetchResponse>}
-   * @private
-   */
-  fetchAmpCors_(input, init = {}) {
-    // Do not append __amp_source_origin if explicitly disabled.
-    if (init.ampCors !== false) {
-      input = this.getCorsUrl(this.win, input);
-    } else {
-      init.requireAmpResponseSourceOrigin = false;
```

Comment:
@jridgewell I have deleted code for `requireAmpResponseSourceOrigin` please review these conditions once